### PR TITLE
[Merged by Bors] - feat: port AlgebraicGeometry.ProjectiveSpectrum.StructureSheaf

### PR DESCRIPTION
**Pull request by joelriou**
---

It compiles, but there is a problem of scope for a local notation discussed below.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/ProjectiveSpectrum/StructureSheaf.lean [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+variable {R A : Type _}
+
+variable [CommRing R] [CommRing A] [Algebra R A]
+
+variable (𝒜 : ℕ → Submodule R A) [GradedAlgebra 𝒜]
+
+-- porting note: this should be "local notation", but as "local notation", `at x` does not work
+-- this should be fixed before merging...
+set_option quotPrecheck false in
+notation "at " x => HomogeneousLocalization.AtPrime 𝒜
```
Comment (joelriou) [inline Mathlib/AlgebraicGeometry/ProjectiveSpectrum/StructureSheaf.lean:71]: This should be a local notation, but if I put `local notation`, the notation just does not work. How can we specify the correct scope of this notation?

Comment (mattrobball) [reply Mathlib/AlgebraicGeometry/ProjectiveSpectrum/StructureSheaf.lean]: Made it into `local macro` and it seems to work as intended.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/AlgebraicGeometry/ProjectiveSpectrum/StructureSheaf.lean (hunk 1 above)
Comment (kmill) [reply Mathlib/AlgebraicGeometry/ProjectiveSpectrum/StructureSheaf.lean]: #5758 is a PR to get this to pretty print

The `notation` command can't generate pretty printers if there's dot notation or `variable` variables, but `notation3` is able to at least deal with the latter.

**Pull request closed without merging**